portalnet/src/overlay_service.rs

```diff
@@ -1232,6 +1232,12 @@ where
             OverlayRequestError::AcceptError("unable to find ENR for NodeId".to_string())
         })?;
         let enr = crate::discovery::UtpEnr(node_addr.enr);
+
+        // avoid memory allocation if trace isn't enabled
+        let mut enr_str = String::new();
```

Comment:
I've been taught that this pattern of instantiating a mutable object and then mutating it later should be avoided when possible. Granted that was in python, which had fewer guarantees around type safety.

Reply:
Ok I will just do an if else then which will remove the mut